### PR TITLE
fix: [CDS-84675]: adds new API with no pipelineIdentifier in path parameters

### PR DESCRIPTION
**Pull request by sandeshsalunkhegh**
fix: [CDS-84675]: removes pipelineIdentifier from path parameters, making it an optional field

Issue: pipelineIdentifier and/or name has to be removed from the API params.

Changes: Deprecates the existing API, creates new API to import pipeline from Git without pipelineIdentidfier in the API path parameters.

Testing: End-to-End testing done alongside UI changes mentioned below.

UI related changes here : https://github.com/harness/harness-core-ui/pull/19254
Doc: https://harness.atlassian.net/wiki/spaces/CDNG/pages/21572387209/Import+Pipeline+From+Git+New+API

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-84675]: https://harness.atlassian.net/browse/CDS-84675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56271)

### CODE REVIEW

File: pipeline-service/contracts/src/main/java/io/harness/pms/pipeline/PipelineResource.java [hunk 1]
```diff
@@ -423,7 +423,7 @@ public interface PipelineResource {
       @HeaderParam("Load-From-Cache") @DefaultValue("false") String loadFromCache);
 
   @POST
-  @Path("/import/{pipelineIdentifier}")
```
Comment (sahilhindwani-harness) [inline pipeline-service/contracts/src/main/java/io/harness/pms/pipeline/PipelineResource.java:426]: Please don't change this api, introduce a new api which takes pipelineIdentifier as a query param

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

File: pipeline-service/contracts/src/main/java/io/harness/pms/pipeline/PipelineResource.java (hunk 1 above)
Comment (sandeshsalunkhegh) [reply pipeline-service/contracts/src/main/java/io/harness/pms/pipeline/PipelineResource.java]: Added an overload for the API with different signature.

**Comment by sandeshsalunkhegh** [comment]
trigger ut3, ut4, ut7, messagecheck

**Comment by sandeshsalunkhegh** [comment]
trigger ut7

**Review by SudarshanPurohit — COMMENTED** [review]
Reviewed the API documentation, looks good. Thank you.

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/contracts/src/main/java/io/harness/pms/pipeline/PipelineResource.java` line 0 at r2](https://reviewable.io/reviews/harness/harness-core/56271#gh-1415002405:-NlgPgDg0xTvrUykwN71:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/1abf26bde425cc0b6c73f2753da317d0575eaea8/pipeline-service/contracts/src/main/java/io/harness/pms/pipeline/PipelineResource.java#L0)):*
<details><summary><i>Previously, sandeshsalunkhegh (Sandesh Salunkhe) wrote…</i></summary><blockquote>

Added an overload for the API with different signature.
</blockquote></details>

Done

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56271)* status: Some of the checks are too old. 3 file(s) to review. 1 of 2 approvals obtained.

**Review by adivishy1 — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56271)* status: Some of the checks are too old. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java` line 838 at r4](https://reviewable.io/reviews/harness/harness-core/56271#-Nm5BZoB6QG1jVEEn7Yv:-Nm5BZoB6QG1jVEEn7Yw:blxgilc) ([raw file](https://github.com/harness/harness-core/blob/1d502f31b36b8c59faef6336e62af3243f3d1ebd/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java#L838)):*
> ```Java
>     String pipelineVersion = pipelineVersion(accountId, importedPipelineYAML);
>     YamlField pipelineRootYamlField = YamlUtils.readYamlTree(importedPipelineYAML).getNode().getField("pipeline");
>     if (EmptyPredicate.isEmpty(pipelineIdentifier)
> ```

nit: we can remove the redundant import EmptyPredicate, we can directly use isEmpty()

**Comment by sandeshsalunkhegh** [comment]
trigger ut9, review

**Comment by sandeshsalunkhegh** [comment]
trigger ut5, ut6, ut7, ut8, ut9

**Comment by sandeshsalunkhegh** [comment]
trigger compile

**Pull request merged**